src/super_context.ml

```diff
@@ -630,10 +626,13 @@ module Action = struct
       ; ddeps     = String.Map.empty
       }
     in
-    let expand loc key var = function
-      | Some ("exe"     , s) -> Some (path_exp (map_exe (Path.relative dir s)))
-      | Some ("path"    , s) -> Some (path_exp          (Path.relative dir s) )
-      | Some ("bin"     , s) -> begin
+    let expand var =
+      let loc = String_with_vars.Var.loc var in
+      let key = String_with_vars.Var.name var in
```

Comment:
The `key` needs to take the payload into account. It is used for the second step, once things like `%{read:...}` have been evaluated. If we only store the name we can't distinguish between `%{read:a}` and `%{read:b}` and they will both be replaced by the same thing.

Reply:
`String_with_vars.Var.name` concatenates the name and the payload. So it would return something like `foo:bar` where both fields are non blank. I should probably give this a better name. How about `full_name` or `payload_name`?

Reply:
Ah, I missed that. `full_name` looks good